spec/factories/document_type_factory.rb

```diff
@@ -28,7 +27,7 @@
     after(:build) do |document_type|
       DocumentType.all << document_type
       Supertype.all.first.document_types << document_type
-      en = { document_types: { document_type.id.to_sym => { description: "..." } } }
+      en = { document_types: { document_type.id.to_sym => { label: document_type.id, description: "..." } } }
```

Comment:
Could we use the defaulting mechanism to specify a "fallback" label? I hesitate here because the label is definitely required, but it's tempting because it reduces the coupling between the models and the locales. The central question here is whether one should be able to add a new document type without any locale information; while the upcoming work on tests should discourage that, functionally it seems reasonable. What do you think?

Reply:
If we did provide a fallback label, it would still be a hardcoded string (unless YAML can do something clever I don't know about!). This would break a number of tests where a document type has to be chosen, as over the course of the test suite various document types build up, so our test would be trying to pick from this:

```html
<select>
  <option value="the-fallback-label">the-fallback-label</option>
  <option value="the-fallback-label">the-fallback-label</option>
  <option value="the-fallback-label">the-fallback-label</option>
  <option value="the-fallback-label">the-fallback-label</option>
  ...
</select>
```

...and then RSpec complains about an ambiguous match.

Reply:
Clearing the dynamically created document types is something we should look at in a future PR.

Reply:
We've decided to keep 'label' in config/document_types.yml - there will be a commit that encapsulates this.